src/modules/inventory/include/inventory.hpp

```diff
@@ -31,7 +31,7 @@ class Inventory {
         void Stop();
         Co_CommandExecutionResult ExecuteCommand(const std::string query);
         const std::string& Name() const { return m_moduleName; };
-        void SetMessageQueue(const std::shared_ptr<IMultiTypeQueue> queue);
+        void SetPushMessageFunction(const std::function<int(Message)>& pushMessage);
```

Comment:
The return type could also be a `bool` right? Maybe the idea is to support error codes in the future? I would still then define an enum class for that, though in the meantime I'd prefer to narrow it down.

Reply:
The return type is `<int>` since the _push_ function of the queue has that return type. In the future we should define error codes.